cloudtools/submit.py

```diff
@@ -64,15 +93,102 @@ def main(args):
         '--properties={}'.format(properties)
     ]
 
+    arg_array = []
     # append arguments to pass to the Hail script
     if args.args:
+        arg_array = args.args.split()
         cmd.append('--')
-        for x in args.args.split():
+        for x in arg_array:
             cmd.append(x)
 
     # print underlying gcloud command
     print('gcloud command:')
     print(' '.join(cmd[:6]) + ' \\\n    ' + ' \\\n    '.join(cmd[6:]))
 
     # submit job
-    call(cmd)
+    #call(cmd)
+        
+    # upload the hail script to this dataproc staging bucket
+    script_name = os.path.basename(args.script)            
+    upload_blob(cluster_staging_bucket, args.script, script_name)
+                           
+    job_details = {
+        'projectId': args.project,
+        'job': {
+            'placement': {
+                'clusterName': args.name
+            },
+            # https://cloud.google.com/dataproc/docs/reference/rest/v1/projects.regions.jobs#pysparkjob
+            'pysparkJob': {
+                "mainPythonFileUri": "gs://{}/{}".format(cluster_staging_bucket, script_name),
+                "args": arg_array,
+                "pythonFileUris": [
+				    "gs://hail-common/pyhail-hail-is-master-{}.zip".format(hash_name),
+				    "gs://{}/{}".format(cluster_staging_bucket, script_name)
+                ],
+                "jarFileUris": [
+				    jar_path
+                ],
+                "fileUris": [
+ 				    uploaded_file_paths
+                 ],
+                 "archiveUris": [
+				    ["gs://{}/{}".format(cluster_staging_bucket, zip_path)]
+                ],
+                "properties": {
+				    "spark.driver.extraClassPath":"./{}".format(hail_jar),
+				    "spark.executor.extraClassPath":"./{}".format(hail_jar)
+                },
+                # "loggingConfig": {
+# 				    object(LoggingConfig)
+#                 },
+            } 
+        }
+    }
+    
+    if args.properties:
+        # convert the properties format to a dict that can be pushed into the job json
+        properties_json = job_details['job']['pysparkJob']['properties']
+        # note we join the properties and then split because properties before this point is a mix
+        # of a list of individual properties and the comma separated string that comes in from args
+        for k,v in dict(s.split('=') for s in args.properties.split(",")).iteritems():
+            properties_json[k] = v
+    
+    
+    result = dataproc.projects().regions().jobs().submit(
+        projectId=args.project,
+        region=args.region,
+        body=job_details).execute()
+    job_id = result['reference']['jobId']
+    
+    job_url = "https://console.cloud.google.com/dataproc/jobs/{}?project={}&region={}".format(job_id, args.project, args.region)
+    
+    print('Submitted job ID {}.  See {} for more details.'.format(job_id, job_url))
+
+def wait_for_job(dataproc, project, region, job_id):
```

Comment:
This is actually not used right now - it submits the job and doesn't wait, but this would be used if we wanted to monitor it's status